### PR TITLE
Terminal api improvments

### PR DESCRIPTION
**Pull request by AlexandrosMor**
#17 Adding async/await to IClient interface 
Make numeric values nullable for nexo models. Numeric values are not serialized if they are not populated.

### CODE REVIEW

File: Adyen.EcommLibrary/Service/ServiceResource.cs [hunk 1]
```diff
@@ -24,5 +25,13 @@ public string Request(string json)
             return clientInterface.Request(Endpoint, json, config);
         }
 
+
+        public async Task<string> RequestAsync(string json)
+        {
+            var clientInterface = _abstractService.Client.HttpClient;
+            var config = _abstractService.Client.Config;
+          
+            return await clientInterface.RequestAsync(Endpoint, json, config,false);
```
Comment (lancergr) [inline Adyen.EcommLibrary/Service/ServiceResource.cs]: let's remove await

File: Adyen.EcommLibrary/Service/Payment.cs [hunk 2]
```diff
@@ -33,13 +34,30 @@ public PaymentResult Authorise(PaymentRequest paymentRequest)
             return paymentResult;
         }
 
+        public async Task<PaymentResult> AuthoriseAsync(PaymentRequest paymentRequest)
+        {
+            PaymentResult paymentResult = null;
```
Comment (lancergr) [inline Adyen.EcommLibrary/Service/Payment.cs]:
```suggestion
            
                var jsonRequest = Util.JsonOperation.SerializeRequest(paymentRequest);
                var jsonResponse = await _authorise.RequestAsync(jsonRequest);
                return JsonConvert.DeserializeObject<PaymentResult>(jsonResponse);
```

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

Comment (AlexandrosMor) [reply Adyen.EcommLibrary/Service/Payment.cs]: If we remove this try catch we need to remove it for Authorise too. I would suggest 
try { ... }catch(Exception ex) { throw ex; } similar to Authorise and Authorise3D

File: Adyen.EcommLibrary/Service/ServiceResource.cs [hunk 3]
```diff
@@ -26,12 +26,12 @@ public string Request(string json)
...
 
 
-        public async Task<string> RequestAsync(string json)
+        public Task<string> RequestAsync(string json)
         {
             var clientInterface = _abstractService.Client.HttpClient;
             var config = _abstractService.Client.Config;
           
-            return await clientInterface.RequestAsync(Endpoint, json, config,false);
+            return clientInterface.RequestAsync(Endpoint, json, config, false);
```
Comment (AlexandrosMor) [inline Adyen.EcommLibrary/Service/ServiceResource.cs]: Hey @lancergr, The Await Async is removed for RequestAsync

**Pull request merged**